Mathlib/CategoryTheory/Sites/IsSheafFor.lean

```diff
@@ -700,4 +700,102 @@ theorem isSheafFor_subsieve (P : Cᵒᵖ ⥤ Type w) {S : Sieve X} {R : Presieve
   isSheafFor_subsieve_aux P h (by simpa using trans (𝟙 _)) fun Y f _ => (trans f).isSeparatedFor
 #align category_theory.presieve.is_sheaf_for_subsieve CategoryTheory.Presieve.isSheafFor_subsieve
 
+section Arrows
+
+variable {B : C} {I : Type*} {X : I → C} (π : (i : I) → X i ⟶ B) (P)
+
+/--
+A more explicit version of `FamilyOfElements.Compatible` for a `Presieve.ofArrows`.
+-/
+def Arrows.Compatible (x : (i : I) → P.obj (op (X i))) : Prop :=
+  ∀ i j Z (gi : Z ⟶ X i) (gj : Z ⟶ X j), gi ≫ π i = gj ≫ π j →
+    P.map gi.op (x i) = P.map gj.op (x j)
+
+lemma FamilyOfElements.isAmalgamation_iff_ofArrows (x : FamilyOfElements P (ofArrows X π))
+    (t : P.obj (op B)) :
+    x.IsAmalgamation t ↔ ∀ (i : I), P.map (π i).op t = x _ (ofArrows.mk i) := by
+  refine ⟨fun h i ↦ h _ (ofArrows.mk i), fun h Y f hf ↦ ?_⟩
+  obtain ⟨i, rfl, hi⟩ := ofArrows_surj π f hf
+  obtain rfl : f = π i := by simpa using hi
+  exact h i
+
+namespace Arrows.Compatible
+
+variable {x : (i : I) → P.obj (op (X i))} (hx : Compatible P π x)
+variable {P π}
+
+theorem exists_familyOfElements :
+    ∃ (x' : FamilyOfElements P (ofArrows X π)), ∀ (i : I), x' _ (ofArrows.mk i) = x i := by
+  use (fun Y f hf ↦ P.map (eqToHom (ofArrows_surj π f hf).choose_spec.choose.symm).op (x _))
+  intro i
+  specialize hx i (ofArrows_surj π (π i) (ofArrows.mk _)).choose (X i) (𝟙 _)
+    (eqToHom (ofArrows_surj π (π i) (ofArrows.mk _)).choose_spec.choose.symm) ?_
+  · simp [← (ofArrows_surj π (π i) (ofArrows.mk _)).choose_spec.choose_spec]
+  rw [← hx]
+  simp
+
+/--
+A `FamilyOfElements` associated to an explicit family of elements.
+-/
+noncomputable
+def familyOfElements {x : (i : I) → P.obj (op (X i))}
+    (hx : Compatible P π x) : FamilyOfElements P (ofArrows X π) :=
+  (exists_familyOfElements hx).choose
+
+@[simp]
+theorem familyOfElements_ofArrows_mk (i : I) :
+    hx.familyOfElements _ (ofArrows.mk i) = x i :=
+  (exists_familyOfElements hx).choose_spec _
+
+theorem familyOfElements_compatible {x : (i : I) → P.obj (op (X i))}
+    (hx : Compatible P π x) : hx.familyOfElements.Compatible := by
+  intro Y₁ Y₂ Z g₁ g₂ f₁ f₂ h₁ h₂ hgf
+  cases' h₁ with i
```

Comment:
Can you use `case` instead of `cases'`? Not very important, but sooner or later we will get rid of `cases'`.